### ISSUE TITLE
`Cannot read property 'transform' of undefined` thrown when use precomputeStyle

### ISSUE THREAD

**Issue by tdzl2003** [facebook/react-native#2999]
I read the code 'precomputeStyle.js', at line 66:

```
function _processor(key: string, validAttributes: Object) {
  var process = validAttributes[key] && validAttributes[key].process;
  if (!process) {
    process =
      ReactNativeStyleAttributes[key] &&
      ReactNativeStyleAttributes[key].process;
  }

  return process;
}
```

when the validAttribute is null, the error is thrown.

Should this line

```
  var process = validAttributes[key] && validAttributes[key].process;
```

changed to: 

```
  var process = validAttributes && validAttributes[key] && validAttributes[key].process;
```

?

**Comment by buhe** [comment]
send pr?

**Comment by ide** [comment]
I haven't seen validAttributes be null. I guess we add an invariant to make the error message be clearer.

**Comment by tdzl2003** [comment]
Maybe third-party components cause this error. If this is a behavior change，should we give warning in current version?
Ide, we really need change log for 0.12 ~~

**Comment by ide** [comment]
Everyone is too busy so we linked to the commit history. Sorry about 0.12. Hopefully 0.13's changelog is helpful.

**Comment by sulliwane** [comment]
@tdzl2003 Could you please tell me where should to replace the line to temporary fix this bug? I can't find precomputeStyle.js anywhere...thank you!

![precompute](https://cloud.githubusercontent.com/assets/1733686/10598685/ad821688-772d-11e5-87c5-b90acc9b89bb.png)
undefined is not an object (evaluating 'validAttributes[key]')

**Comment by tdzl2003** [comment]
As the error message say, in 'precomputeStyle.js'
It's in `node_modules/react-native/Libraries/StyleSheet`

**Comment by sulliwane** [comment]
@tdzl2003 right, didn't find it at first. Thank you!

**Comment by watert** [comment]
seems the precomputeStyle have no default `validAttributes` value, can be added manually to avoid this problem, but seems it's more like a bug.

it works for me to manually pass a second variable to set the `validAttributes` like below:

`let nativeProps = precomputeStyle({ transform: [{rotate: '45deg'}] }, {transform:true});`

**Comment by tdzl2003** [comment]
This issue should already be fixed after 0.15. Closing this.
